.eslintrc.js

```diff
@@ -14,6 +14,44 @@ module.exports = {
     {
       files: ['*.test.ts', '*.test.js'],
       extends: ['@metamask/eslint-config-jest'],
+      rules: {
```

Comment:
I'm guessing your motivation here is to avoid writing doc comments for test helpers. Could you elaborate on why you want to avoid doing that? I find such comments useful for understanding how to use the test helpers, and for understanding what the tests do.

Also, I suspect that `jsdoc/require-jsdoc` is the only rule that needs to be disabled here.  The rest of the rules are for ensuring any JSDoc comments that do exist are complete and formatted correctly. Even if we want to make JSDoc comments optional in tests, we'd still want them to be complete and correctly formatted.

Reply:
So as a little context, I have some helper functions I've added to the tests for GasFeeController. I realize this is a little different from the pattern set by other test files, but for these tests it seemed like a better way to organize things as we have a lot of setup code in this file. I am curious whether we should require people to write documentation for functions that are local to test files. In my view, good documentation explains the "why" behind the code, not the "what", and so the best documentation exists in places where the "why" is not obvious. I may be a little biased, but I think the context/intent behind the functions in `GasFeeController.test.ts` is fairly obvious provided that a good name is being provided for those functions and once you see how they are being used, and it would be difficult to provide more information about such functions that could not already be easily derived from the code. For instance, what could you write about `buildReturnValueForFetchGasEstimates`? If you don't know how GasFeeController works, you'd have to explain how it works (but you can do that elsewhere). If you _do_ know how it works, you can see where it's being used in the test, which is as mock data for a `fetchGasEstimates` function being passed into the controller. If that is difficult to do, then there are probably readability issues with the entire file anyway.

Based on this, my worry is that by requiring documentation across the board in test files, we are adding an obstacle to writing tests. That said, perhaps disabling these ESLint rules is a hasty decision given that we don't have many test files that keep their own helpers. So on that alone I would be willing to take these changes back and wait until we have more of these types of situations.

Reply:
Well, the rules only require documentation for helper functions, not the tests themselves. And it is important that helper functions are clearly understood if the tests are to remain readable. Honestly if these modest documentation requirements discouraged writing helper methods completely that wouldn't be a huge loss for me, because they have a tendency to make tests difficult to understand anyway. Better to have long, verbose tests where all of the setup is laid bare, than one involving an idiosyncratic mock that was created by a "createMockX" function that could be doing god knows what.

Honestly `buildReturnValueForFetchGasEstimates` does seem like an example of a difficult-to-understand helper function (at least devoid of further context - I haven't read this PR yet). What does this mock represent? Why were the specific values used? What does this tell us about the code under test? Does this mock represent an edge case, a base case, a typical one, or?

A short description like "Return an example gas estimate API response featuring a typical distribution of estimate times" would help me a lot as a reader. It would tell me that it was meant to represent a typical gas estimate, that I'd have to think about edge cases differently (e.g. what if the API returned zero? What if the low estimate was higher than the high one? What if all of the estimates were very high or very low, with little variation? What if an error was returned?), and maybe add tests for those or add validation to prevent those cases in the API module.

All of that might be clear from a careful read, but a helper function that has to be read carefully to be understood will not improve readability. WIth a nice concise description, I could hover over the helper method in VSCode and see the description without having to leave the test I'm reading. As opposed to the alternative of switching contexts, carefully analyzing what the mock does and trying to guess why, then trying to keep that all in my head when I switch back and try it recall what the test I was reading was doing.

Reply:
I agree with @mcmire that this is not necessarily the forum for litigating the JSDoc lint rules, but since we are here, I'll put my stake in the ground, too.

Having written some awful test helpers in my day, I have to agree with @Gudahtt here. Writing a test helper function docstring should never take more than a few minutes at most, and with autocomplete in e.g. VS Code (mac: cmd+shift+enter after typing `/** */` immediately above a function or method) it's damn-near trivial.

In general, we are asking whether it's worth maybe 2-10 more minutes per function to make all of our code — tests and otherwise — more readable, for everyone, forever. I wrote a bunch of docstrings for a large class earlier today and I hated every minute of it, but I can't say that it's not worth it. We'll get the time I spent writing them back twelve times over in the next 3-6 months, to say nothing of the long term.

Reply:
Okay thanks for the feedback on this! I'll remove these changes and add a description for this function.